src/Twilio/Security/RequestValidator.cs

```diff
@@ -49,14 +49,20 @@ public bool Validate(string url, IDictionary<string, string> parameters, string
             return SecureCompare(signature, expected);
         }
 
-        public bool Validate(string url, NameValueCollection parameters, string body, string expected)
+        public bool Validate(string url, string body, string expected)
         {
-            return Validate(url, ToDictionary(parameters), body, expected);
-        }
+            var paramString = new Uri(url).Query.TrimStart('?');
+            var bodyHash = "";
+            foreach (var param in paramString.Split('&'))
```

Comment:
I guess the only .NET helper function for this is in `System.Web`? it seems nuts that `Uri` can't do this for you. :(

Reply:
Yup. Went the System.Web route, but doesn't work in older versions. :(